ambry-router/src/test/java/com.github.ambry.router/GetManagerTest.java

```diff
@@ -0,0 +1,172 @@
+  @Test
+  public void testSimpleBlobGetBlobInfoSuccess()
+      throws Exception {
+    router = getNonBlockingRouter();
+    setOperationParams(chunkSize);
+    String blobId = router.putBlob(putBlobProperties, putUserMetadata, putChannel).get();
+    BlobInfo blobInfo = router.getBlobInfo(blobId).get();
+    Assert.assertEquals("Blob properties should match", putBlobProperties, blobInfo.getBlobProperties());
+    Assert.assertArrayEquals("User metadata should match", putUserMetadata, blobInfo.getUserMetadata());
+    router.close();
+  }
+
+  /**
+   * Tests get blob info of composite blobs
+   * @throws Exception
+   */
+  @Test
+  public void testCompositeBlobGetBlobInfoSuccess()
+      throws Exception {
+    router = getNonBlockingRouter();
+    setOperationParams(chunkSize * 6 + 11);
+    String blobId = router.putBlob(putBlobProperties, putUserMetadata, putChannel).get();
+    BlobInfo blobInfo = router.getBlobInfo(blobId).get();
+    Assert.assertEquals("Blob properties should match", putBlobProperties, blobInfo.getBlobProperties());
+    Assert.assertArrayEquals("User metadata should match", putUserMetadata, blobInfo.getUserMetadata());
+    router.close();
+  }
+
+  /**
+   * Tests the failure case where poll throws and closes the router. This also tests the case where the GetManager
+   * gets closed with active operations, and ensures that operations get completed with the appropriate error.
+   * @throws Exception
+   */
+  @Test
+  public void testFailureOnAllPollThatSends()
+      throws Exception {
+    router = getNonBlockingRouter();
+    setOperationParams(chunkSize);
+    String blobId = router.putBlob(putBlobProperties, putUserMetadata, putChannel).get();
+    try {
+      mockSelectorState.set(MockSelectorState.ThrowExceptionOnSend);
+      router.getBlobInfo(blobId).get();
+      Assert.fail("operation should have thrown");
+    } catch (ExecutionException e) {
+      RouterException routerException = (RouterException) e.getCause();
+      Assert.assertEquals("Exception received should be router closed error", routerException.getErrorCode(),
+          RouterErrorCode.RouterClosed);
+    }
+  }
+
+  /**
+   * @return Return a {@link NonBlockingRouter} created with default {@link VerifiableProperties}
+   */
+  private NonBlockingRouter getNonBlockingRouter()
+      throws IOException {
+    Properties properties = new Properties();
+    properties.setProperty("router.hostname", "localhost");
+    properties.setProperty("router.datacenter.name", "DC1");
+    properties.setProperty("router.max.put.chunk.size.bytes", Integer.toString(chunkSize));
+    properties.setProperty("router.put.request.parallelism", Integer.toString(requestParallelism));
+    properties.setProperty("router.put.success.target", Integer.toString(successTarget));
+    VerifiableProperties vProps = new VerifiableProperties(properties);
+    router = new NonBlockingRouter(new RouterConfig(vProps), new NonBlockingRouterMetrics(new MetricRegistry()),
```

Comment:
minor: no need to create a new `MetricRegistry` - the cluster map has one already.